src/NPoco/Expressions/SqlExpression.cs

```diff
@@ -1603,13 +1604,13 @@ protected virtual object VisitColumnAccessMethod(MethodCallExpression m)
                     statement = string.Format("lower({0})", expression);
                     break;
                 case "StartsWith":
-                    statement = string.Format("upper({0}) like {1}", expression, CreateParam(args[0].ToString().ToUpper() + "%"));
+                    statement = string.Format("upper({0}) like {1} ESCAPE'{2}'", expression, CreateParam(EscapeParam(args[0]) + "%"), EscapeChar);
                     break;
                 case "EndsWith":
-                    statement = string.Format("upper({0}) like {1}", expression, CreateParam("%" + args[0].ToString().ToUpper()));
+                    statement = string.Format("upper({0}) like {1} ESCAPE'{2}'", expression, CreateParam("%" + EscapeParam(args[0])), EscapeChar);
                     break;
                 case "Contains":
-                    statement = string.Format("upper({0}) like {1}", expression, CreateParam("%" + args[0].ToString().ToUpper() + "%"));
+                    statement = string.Format("upper({0}) like {1} ESCAPE'{2}'", expression, CreateParam("%" + EscapeParam(args[0]) + "%"), EscapeChar);

```

Comment:
We may want to extract the whole Escape part into a virtual method just in case it is different in another sql dialect.